src/engine/racer.rs

```diff
@@ -1,17 +1,17 @@
 //! SemanticEngine implementation for [the racer library](https://github.com/phildawes/racer)
 //!
-use engine::{SemanticEngine, Definition, Context, CursorPosition, Completion};
+use engine::{SemanticEngine, Definition, Context, CursorPosition, Completion, Error};
 
-use racer::core::{Session, FileCache};
-use racer::scopes::{coords_to_point, point_to_coords};
+use racer::core::{Session, FileCache, IndexedSource};
 
 use std::path::Path;
+use std::rc::Rc;
 
 use std::sync::Mutex;
 use regex::Regex;
 
 pub struct Racer<'a> {
-    cache: Mutex<&'a FileCache<'a>>
```

Comment:
I think this can now be `Mutex<FileCache>` and all these references can go away. It used to be that `FileCache` had a `TypedArena` internally which is why the refs and leaking the `FileCache`.